qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -60,13 +60,10 @@ class QuantumKernel(TrainableKernel, BaseKernel):
 
     @deprecate_arguments("0.5.0", {"user_parameters": "training_parameters"})
     @deprecate_function(
-        "The QuantumKernel class has been superseded by the "
+        "The QuantumKernel class is deprecated and has been superseded by the "
         "qiskit_machine_learning.kernels.FidelityQuantumKernel and "
         "qiskit_machine_learning.kernels.TrainableFidelityQuantumKernel classes. "
-        "This class will be deprecated in a future release and subsequently "
-        "removed after that.",
-        stacklevel=3,
-        category=PendingDeprecationWarning,
+        "This class will be removed in a future release.",
```

Comment:
Updated to Terra 0.24 and replaced the deprecators. Since this new decorator does not include :class:, I don't think we should include Sphinx formatting to messages.

Reply:
Thanks for using the new decorators! I appreciate that - it helps to ensure the docs are a good experience.

I agree with Stephen that I recommend using Sphinx cross-references for the `additional_msg` piece. The generated docs are much more valuable that way because users will have a link to the replacement. Save Them A Click. 

We are wording things like this in Terra.

Before:

> Instead, use the ``qiskit_machine_learning.kernels.FidelityQuantumKernel``" and ``qiskit_machine_learning.kernels.TrainableFidelityQuantumKernel`` classes.

After:

> Instead, use :class:`qiskit_machine_learning.kernels.FidelityQuantumKernel` and :class:`qiskit_machine_learning.kernels.TrainableFidelityQuantumKernel`.

That is, we don't repeat the word "classes" because the cross-reference already makes that clear.

I agree that this runtime message is a little more awkward. But in Terra, we decided it was well worth the trade-off so that the docs are more useful.

Reply:
> so for consistency reasons I don't include :class: in the additional message either.

Personally, I don't think that's a very big deal. 

It would be silly to use a cross-reference at the very start of the message because it doesn't make sense to cross-reference the deprecated entity itself. This message will already show up underneath the docs for the class; there is no new page to navigate to.

--

I agree it's perhaps awkward to not be consistent, and also to have `:class:` in the runtime message. But what do we think best optimizes for the end user? No solution is perfect. We are deciding which is worse: the awkwardness of the slight inconsistency & `:class:` syntax in the runtime message vs. having less useful generated docs that require manually navigating around.

For Terra, we decided that it's "worth it" to optimize for the better docs experience.